File: src/client/java/io/github/ricciow/util/message/PagedMessageFactory.java
package io.github.ricciow.util.message;

import io.github.ricciow.PridgeClient;
import net.minecraft.text.Text;
import net.minecraft.text.TextColor;
import org.jetbrains.annotations.Nullable;

import java.util.List;

import static com.mojang.brigadier.arguments.StringArgumentType.getString;
import static com.mojang.brigadier.arguments.StringArgumentType.word;
import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.argument;
import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.literal;

public class PagedMessageFactory {

    static PagedMessage lastPagedMessage;

    /**
     * Creates a paged message, only the last sent paged message will be able to change pages.
     */
    public static void createPagedMessage(List<Text> pages, Text title, TextColor arrowColor, @Nullable TextColor disabledArrowColor) {
        if(lastPagedMessage != null) {
            lastPagedMessage.disablePaging();
        }
        lastPagedMessage = new PagedMessage(pages, title, arrowColor, disabledArrowColor);
    }

    public static void createPagedMessage(List<Text> pages, List<Text> titles, TextColor arrowColor, @Nullable TextColor disabledArrowColor) {
        if(lastPagedMessage != null) {
            lastPagedMessage.disablePaging();
        }
        lastPagedMessage = new PagedMessage(pages, titles, arrowColor, disabledArrowColor);
    }

    public static void initialize() {
        PridgeClient.COMMAND_MANAGER.addCommand(
            literal("pagedmessage")
                .then(argument("direction", word())
                    .executes(commandContext -> {
                        if(lastPagedMessage != null) {
                            if (getString(commandContext, "direction").equals("left")) {
                                lastPagedMessage.lastPage();
                            } else {
                                lastPagedMessage.nextPage();
                            }
                        }
                        return 1;
                    })
                )
        );
    }
}

File: src/main/kotlin/io/github/ricciow/util/message/PagedMessageFactory.kt
package io.github.ricciow.util.message

import net.minecraft.text.Text
import net.minecraft.text.TextColor

object PagedMessageFactory {
    private val pagedMessages = mutableMapOf<Int, PagedMessage>()
    fun getMessageById(id: Int): PagedMessage? {
        return pagedMessages[id]
    }

    /**
     * Creates a paged message, only the last sent paged message will be able to change pages.
     */
    fun createPagedMessage(
        pages: MutableList<Text>,
        title: Text,
        arrowColor: TextColor,
        disabledArrowColor: TextColor,
        prefix: Text?
    ) {
        val pagedMessage = PagedMessage(pages, title, arrowColor, disabledArrowColor, prefix)
        pagedMessages[pagedMessage.id] = pagedMessage
    }

    fun createPagedMessage(
        pages: MutableList<Text>,
        titles: MutableList<Text>,
        arrowColor: TextColor,
        disabledArrowColor: TextColor,
        prefix: Text?
    ) {
        val pagedMessage = PagedMessage(pages, titles, arrowColor, disabledArrowColor, prefix)
        pagedMessages[pagedMessage.id] = pagedMessage
    }
}